### PR TITLE
feat: change LD annotation for PICS fine-mapping to use major ancestry

### PR DESCRIPTION
**Pull request by vivienho**
## ✨ Context

LD annotation for PICS fine-mapping previously combined multiple ancestries. In this PR this is changed so that only the major ancestry is used.

## 🛠 What does this PR implement

Added `_get_major_population` and `_calculate_r2_major` functions and their corresponding tests.

~~All mentions of `r2Overall` were changed to `r2Major` which is now a better name for this field.~~
`r2Overall` is kept as we may come back to it in the future.

Functions (`_add_population_size` and `_calculate_weighted_r_overall`) and corresponding tests that are no longer needed were removed.

## 🚦 Before submitting

- [x] Do these changes cover one single feature (one change at a time)?
- [x] Did you read the [contributor guideline](https://opentargets.github.io/gentropy/development/contributing/#contributing-checklist)?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you make sure there is no commented out code in this PR?
- [x] Did you follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standards in PR title and commit messages?
- [x] Did you make sure the branch is up-to-date with the `dev` branch?
- [x] Did you write any new necessary tests?
- [x] Did you make sure the changes pass local tests (`make test`)?
- [x] Did you make sure the changes pass pre-commit rules (e.g `poetry run pre-commit run --all-files`)?

### CODE REVIEW

File: src/gentropy/assets/schemas/study_locus.json [hunk 1]
```diff
@@ -158,7 +158,7 @@
             },
             {
               "metadata": {},
-              "name": "r2Overall",
+              "name": "r2Major",
```
Comment (addramir) [inline src/gentropy/assets/schemas/study_locus.json:161]: I think renaming this column is bad idea. We may return to overall in the future. Moreover changing schema now is very bad idea. I suggest use r2Overall.

Comment (vivienho) [reply src/gentropy/assets/schemas/study_locus.json:161]: @DSuveges mentioned this as well, currently reverting those changes

**Review by DSuveges — APPROVED** [review]
Thank you for reverting the column names. Everything makes total sense, however I havent' had time to run the code. Good job handling cases when there are multiple major populations with the same relativeSampleSizes.

File: src/gentropy/method/ld.py [hunk 2]
```diff
@@ -19,60 +20,64 @@ class LDAnnotator:
...
-                    + f.coalesce(
-                        f.pow(y["r"], 2) * y["relativeSampleSize"], f.lit(0.0)
-                    ),  # we use coalesce to avoid problems when r/relativeSampleSize is null
-                ).alias("r2Overall"),
-            ),
+        major_population_size = ordered_populations["relativeSampleSize"][0]
+        major_populations = f.filter(
+            ordered_populations,
+            lambda x: x["relativeSampleSize"] == major_population_size
+        )
```
[the remark this answers is not in the document]
Comment (vivienho) [reply src/gentropy/method/ld.py:39]: Yes, that's correct.

**Pull request merged**